### PR TITLE
add numbers to too much slippage and too expensive

### PR DESCRIPTION
**Pull request by coburncoburn**
This patch adds more information to some difficult to debug reverts.

### CODE REVIEW

File: src/Paycall.sol [hunk 1]
```diff
@@ -17,7 +17,7 @@ contract Paycall {
 
     error BadPrice();
     error InvalidCallContext();
-    error TransactionTooExpensive();
+    error TransactionTooExpensive(uint256 expectedMaxCost, uint256 actualMaxCost);
```
Comment (kevincheng96) [inline src/Paycall.sol:20]: Similar change should be made to `Quotecall` as well

File: src/lib/DeFiScriptErrors.sol [hunk 2]
```diff
@@ -10,5 +10,5 @@ library DeFiScriptErrors {
     error InvalidInput();
     error TransferFailed(bytes data);
     error ApproveAndSwapFailed(bytes data);
-    error TooMuchSlippage();
+    error TooMuchSlippage(uint256 actualBuyAmount, uint256 expectedBuyAmount);
```
Comment (kevincheng96) [inline src/lib/DeFiScriptErrors.sol:13]: nit: we should flip the order to be consistent with `TransactionTooExpensive(uint256 expectedMaxCost, uint256 actualMaxCost)`, which has expected first, then actual

**Comment by kevincheng96** [comment]
Looks like there is a failing unit test

**Review by fluffywaffles — COMMENTED** [review]
Agree with ⛑️  that we should stick to `(expected, actual)`.

Overall LGTM but will hold ✅ for ⛑️ 's comments.

File: src/lib/DeFiScriptErrors.sol (hunk 2 above)
Comment (coburncoburn) [reply src/lib/DeFiScriptErrors.sol]: doh, yes

**Comment by coburncoburn** [comment]
> Looks like there is a failing unit test

weird this test passes locally and in fact is explicitly the test that I changed...

File: .github/workflows/test.yml
Comment (kevincheng96) [inline .github/workflows/test.yml:1]: Should we revert these changes?

**Review by kevincheng96 — APPROVED** [review]
Look good. Weird that snapshot is failing though. Maybe it needs to be re-gen'ed again?

File: .github/workflows/test.yml
Comment (coburncoburn) [reply .github/workflows/test.yml:1]: yeah the forge fmt changes are too much for me to stomach today

**Review by kevincheng96 — APPROVED** [review]
